benchmark/sirun/runall.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -e
+
+DIRS=($(ls -d */ | sed 's:/$::')) # Array of subdirectories
+CWD=$(pwd)
+
+function cleanup {
+  for D in "${DIRS[@]}"; do
+    rm -f "${CWD}/${D}/meta-temp.json"
+  done
+}
+
+trap cleanup EXIT
+
+if test -f ~/.nvm/nvm.sh; then
+  source ~/.nvm/nvm.sh
+else
+  source /usr/local/nvm/nvm.sh
+fi
+
+# run each test in parallel for a given version of Node.js
+# once all of the tests have complete move on to the next version
+
+TOTAL_CPU_CORES=$(nproc 2>/dev/null || echo "24")
+
+export CPU_AFFINITY="${CPU_START_ID:-$TOTAL_CPU_CORES}" # Benchmarking Platform convention
+
+nvm install $MAJOR_VERSION # provided by each benchmark stage
+export VERSION=`nvm current`
+export ENABLE_AFFINITY=true
+
+echo "using Node.js ${VERSION}"
+CPU_AFFINITY="${CPU_START_ID:-$TOTAL_CPU_CORES}" # reset for each node.js version
+SPLITS=${SPLITS:-1}
+GROUP=${GROUP:-1}
+
+BENCH_COUNT=0
+for D in "${DIRS[@]}"; do
+  cd "${D}"
+  variants="$(node ../get-variants.js)"
+  for V in $variants; do BENCH_COUNT=$(($BENCH_COUNT+1)); done
+  cd ..
+done
+
+echo "BENCH_COUNT: ${BENCH_COUNT}"
+GROUP_SIZE=$(($(($BENCH_COUNT+$SPLITS-1))/$SPLITS)) # round up
+
+BENCH_INDEX=0
+BENCH_END=$(($GROUP_SIZE*$GROUP))
+BENCH_START=$(($BENCH_END-$GROUP_SIZE))
+
+if [[ ${GROUP_SIZE} -gt 24 ]]; then
+  echo "Group size ${GROUP_SIZE} is larger than available number of CPU cores on Benchmarking Platform machines (${TOTAL_CPU_CORES} cores)"
+  exit 1
```

Comment:
Instead of exiting, should we not just start the additional variations as soon as a core finished with its results? That way it would just wait with the startup and never fail due to too many variations. It would just have to be a specific core that is predetermined to make sure the benchmark is run with the same core in both code variations (CPU cores may have different performance profiles).

Reply:
I'll change this script a bit, as in dd-trace-js we are executing it in different groups in parallel and posting the message with regression after that. It would be simpler (longer if we have a lot of benchmarks in this repo) if we do something similar to what you are suggesting.

I'll change it to calculate and execute the groups in this script, and execute the groups one after one, but in parallel as expected in each group. 

_(anyway, we are far from the max group size in the CI, which is 24 (48 cores) and we have 2 tests)_